test/XmlHttpRequestConnection.tests.js

```diff
@@ -145,6 +145,80 @@ define(function (require) {
 
         });
 
+        describe("is correctly using XmlHttpRequest and runs attached callbacks while performing:", function (){
+            beforeEach(function (){
+                mockXMLHttpRequest.prototype.send = function (body) {
+                    this.readyState = 4;
+                    this.status = 200;
+                    this.onloadstart();
+                    this.onload();
+                    this.onloadend();
+                    this.onprogress();
+                    this.ontimeout();
+                    this.onerror();
+                    this.onabort();
+                    this.upload.onloadstart();
+                    this.upload.onload();
+                    this.upload.onloadend();
+                    this.upload.onprogress();
+                    this.upload.ontimeout();
+                    this.upload.onerror();
+                    this.upload.onabort();
+                    this.onreadystatechange();
+                };
+                mockXMLHttpRequest.prototype.upload = {};
+                spyOn(mockXMLHttpRequest.prototype, 'send').andCallThrough();
+
+                window.XMLHttpRequest = (function () { return mockXMLHttpRequest; }());
+
+                connection = new XmlHttpRequestConnection();
+            });
+
+            it("execute call and invokes custom request event callbacks", function () {
```

Comment:
One more thing. Everytime we do a request we are setting up a new XHR object hence why I don't see a point in using `addEventListener` methods in this case when you can apply one callback to one event. If someone wants to add many callbacks to a specific event then something went wrong in implementing the functionality. 
The other thing is we are not reusing XHR objects. They will be garbage collected when they finish handling request.